admin-dev/themes/new-theme/js/stock-page/components/products/product/spinner.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+  <form class="qty text-xs-right" :class="classObject" v-on:mouseover="focusIn" v-on:mouseleave="focusOut($event)">
+    <input @keyup="onKeyup($event.target.value)" v-on:focus="focusIn" v-on:blur="focusOut($event)" :id="id" class="edit-qty" name="qty" v-model="qty" placeholder="0" >
+    <transition name="fade">
+      <button v-if="isActive" class="check-button" v-on:click="sendQty($event)"><i class="material-icons">check</i></button>
+    </transition>
+  </form>
+</template>
+
+<script>
+  export default {
+    props: ['product'],
+    mounted() {
+      let self = this;
+      $(`#${this.id}`).spinner({
+        max: 999999999,
+        min: -999999999,
+        spin(event, ui) {
+          self.value = ui.value;
+          self.isEnabled = !!self.value;
+        }
+      });
+    },
+    computed: {
+      qty () {
+        if(!this.product.qty) {
+          this.isActive = this.isEnabled = false;
+          this.value = this.product.qty = null;
+        }
+        return this.product.qty;
+      },
+      id () {
+        return `qty-${this.product.product_id}-${this.product.combination_id}`;
+      },
+      classObject() {
+        return {
+          active: this.isActive,
+          disabled: !this.isEnabled
+        }
+      }
+    },
+    data() {
+      return {
+        value: null,
+        isActive: false,
+        isEnabled: false
+      }
+    },
+    watch: {
+      value(val) {
+        this.$store.commit('UPDATE_PRODUCT_QTY', {
+          product_id: this.product.product_id,
+          combination_id: this.product.combination_id,
+          delta: val
+        });
+      }
+    },
+    methods: {
+      onKeyup(val) {
+        this.value = val;
+        if(this.value) {
+          this.isActive = this.isEnabled = true;
+        }
+      },
+      focusIn() {
+        this.isActive = true;
+      },
+      focusOut(event) {
+        if(!$(event.relatedTarget).hasClass('check-button') && !this.value) {
+          this.isActive = false;
+        }
+        this.isEnabled = !!this.value;
+      },
+      sendQty(event) {
+        let apiRootUrl = data.apiRootUrl.replace(/\?.*/,'');
+        let apiEditProductsUrl = `${apiRootUrl}/product/${this.product.product_id}`
+        let apiEditCombinationsUrl = `${apiRootUrl}/product/${this.product.product_id}/combination/${this.product.combination_id}`;
+        let postUrl = this.product.combination_id ? apiEditCombinationsUrl : apiEditProductsUrl;
+
+        event.preventDefault();
```

Comment:
@mickaelandrieu oh, i think that you can use it but you also can use it as a shorthand in a element:
`<form v-on:submit.prevent="onSubmit"></form>`